packages-common/pluggable-widgets-generator/generators/app/index.js

```diff
@@ -0,0 +1,284 @@
+const { join } = require("path");
+const pkg = require(join(__dirname, "/../../package.json"));
+const { mkdir } = require("fs").promises;
+const Generator = require("yeoman-generator");
+
+const promptTexts = require("./lib/prompttexts.js");
+const text = require("./lib/text.js");
+const utils = require("./lib/utils.js");
+
+const widgetSrcFolder = "src/components/";
+
+const banner = text.getBanner(pkg);
+
+class MxGenerator extends Generator {
+    constructor(args, opts) {
+        super(args, opts);
+
+        /**
+         * Test if a widget folder was passed by argument and try to create the new folder
+         */
+        if (args.length > 0) {
+            const dir = args.map(arg => arg.replace(/(^|\s)\S/g, l => l.toLowerCase())).join("-");
+            const name = args.map(arg => arg.replace(/(^|\s)\S/g, l => l.toUpperCase())).join("");
+            if (dir) {
+                this.dir = dir;
+                this.widgetParamName = name;
+            }
+        }
+    }
+
+    async initializing() {
+        this.FINISHED = false;
```

Comment:
Can we achieve the same by throwing an error in the below cases? It also will probably make the generator exit with the correct non-zero code